cmd/gosh/main.go

```diff
@@ -15,9 +17,21 @@ import (
 var parser *syntax.Parser
 
 func main() {
+	var command bool
+	flag.BoolVar(&command, "c", false, "command to be executed")
 	flag.Parse()
 	parser = syntax.NewParser()
 
+	if command {
+		for _, c := range flag.Args() {
```

Comment:
Are you positive this is how `-c` works? Reading bash's man page just confused me. Saner shells seem to use `-c string` instead of arguments, which is how I assumed it worked.

In any case, your current implementation seems wrong:

```
$ bash -c 'echo foo' 'echo bar'
foo
```